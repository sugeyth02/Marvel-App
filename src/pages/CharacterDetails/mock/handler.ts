import { rest } from 'msw';

const baseURL = 'https://gateway.marvel.com/v1/public/';

export const handlers = [
  rest.get(`${baseURL}characters/1009149`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          results: [
            {
              id: 1009149,
              name: 'Abyss',
              modified: '2014-04-29T14:10:43-0400',
              thumbnail: {
                path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
              },
              comics: { items: [{ name: 'Uncanny X-Men (1963) #402' }] },
              stories: { items: [{ name: 'A Beginning' }] },
            },
          ],
        },
      })
    );
  }),
];