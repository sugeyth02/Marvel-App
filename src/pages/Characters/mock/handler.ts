import { rest } from 'msw';

const baseURL = 'https://gateway.marvel.com/v1/public/';

export const handlers = [
  rest.get(`${baseURL}characters`, (req, res, ctx) => {
    const name = req.url.searchParams.get('nameStartsWith');
    if (name === 'MARVEL') {
      return res(
        ctx.status(200),
        ctx.json({
          data: {
            count: 1560,
            results: [
              {
                id: 1011350,
                name: 'Marvel Apes',
                thumbnail: {
                  path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
                },
              },
              {
                id: 1009427,
                name: 'Marvel Boy',
                thumbnail: {
                  path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available',
                },
              },
              {
                id: 1011002,
                name: 'Marvel Zombies',
                thumbnail: {
                  path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available',
                },
              },
            ],
          },
        })
      );
    }
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          count: 1560,
          results: [
            {
              id: 1011334,
              name: '3-D Man',
              thumbnail: {
                path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
              },
            },
            {
              id: 1011170,
              name: 'Alain',
              thumbnail: {
                path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available',
              },
            },
          ],
        },
      })
    );
  }),

  rest.get(`${baseURL}stories`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          results: [
            {
              id: 602,
              title: 'Cover #602',
            },
            {
              id: 631,
              title: '1 of 5 - Golgotha',
            },
          ],
        },
      })
    );
  }),

  rest.get(`${baseURL}comics`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          results: [
            {
              id: 36594,
              title:
                'DEADPOOL VOL. 7: SPACE ODDITY PREMIERE HC (Trade Paperback)',
            },
            {
              id: 41161,
              title: 'Shame Itself (2011) #1',
            },
          ],
        },
      })
    );
  }),
];
