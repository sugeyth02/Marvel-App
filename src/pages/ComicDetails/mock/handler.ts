import { rest } from 'msw';

const baseURL = 'https://gateway.marvel.com/v1/public/';

export const handlers = [
  rest.get(`${baseURL}comics/1886`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          results: [
            {
              id: 1886,
              title:
                'Official Handbook of the Marvel Universe (2004) #12 (SPIDER-MAN)',
              variantDescription: 'SPIDER-MAN',
              modified: '-0001-11-30T00:00:00-0500',
              thumbnail: {
                path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/4bc64020a4ccc',
              },
              characters: { items: [{ name: 'Apocalypse' }] },
              stories: { items: [{ name: 'Cover #4430' }] },
            },
          ],
        },
      })
    );
  }),
];
