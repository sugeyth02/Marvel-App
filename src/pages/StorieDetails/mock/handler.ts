import { rest } from 'msw';

const baseURL = 'https://gateway.marvel.com/v1/public/';

export const handlers = [
  rest.get(`${baseURL}stories/28`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        data: {
          results: [
            {
              id: 11,
              title: 'Interior #11',
              type: 'story',
              modified: '1969-12-31T19:00:00-0500',
              comics: {
                items: [{ name: 'Captain Britain Vol. I (Trade Paperback)' }],
              },
            },
          ],
        },
      })
    );
  }),
];
