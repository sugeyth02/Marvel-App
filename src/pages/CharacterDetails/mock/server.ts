import { setupServer } from 'msw/node';
import { handlers } from './handler';
import { handlers as characterHandler } from '../../Characters/mock/handler';

const server = setupServer(...handlers,...characterHandler);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
export default server;
