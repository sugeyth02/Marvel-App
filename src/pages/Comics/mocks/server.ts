import { setupServer } from 'msw/node';
import { handlers } from './handler';


const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
export default server;
