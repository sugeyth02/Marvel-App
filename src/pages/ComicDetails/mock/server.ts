import { setupServer } from 'msw/node';
import { handlers } from './handler';
import { handlers as comicsHandler } from '../../Comics/mocks/handler';

const server = setupServer(...handlers, ...comicsHandler);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
export default server;
