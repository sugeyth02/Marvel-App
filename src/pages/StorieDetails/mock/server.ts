import { setupServer } from 'msw/node';
import { handlers } from './handler';
import { handlers as storiesHandler } from '../../Stories/mocks/handler';

const server = setupServer(...handlers, ...storiesHandler);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
export default server;
