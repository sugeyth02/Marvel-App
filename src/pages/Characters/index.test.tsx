import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { store } from './../../redux/store';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';
import './mock/server';
import Router from '../../components/Router';

describe('Testing Characters list view', () => {
  beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  const renderComponent = () =>
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/characters']}>
          <Router />
        </MemoryRouter>
      </Provider>
    );

  test('should renders a list of characters initially', async () => {
    // Render
    const { container } = renderComponent();

    // Assertions
    await waitFor(
      () => {
        expect(screen.queryByText('3-D Man')).toBeInTheDocument();
        expect(screen.queryByText('Alain')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
      },
      { timeout: 3000 }
    );

    const comics = await screen.findByLabelText(/Comics:/i);
    userEvent.click(comics);

    await waitFor(() => {
      expect(
        screen.queryByText(
          'DEADPOOL VOL. 7: SPACE ODDITY PREMIERE HC (Trade Paperback)'
        )
      ).toBeInTheDocument();
      expect(screen.queryByText('Shame Itself (2011) #1')).toBeInTheDocument();
    });

    const stories = await screen.findByLabelText(/Stories:/i);
    userEvent.click(stories);

    await waitFor(() => {
      expect(screen.queryByText('Cover #602')).toBeInTheDocument();
      expect(screen.queryByText('1 of 5 - Golgotha')).toBeInTheDocument();
    });
  });

  test('should search a character', async () => {
    renderComponent();
    await screen.findByText('3-D Man');
    const searchBar = screen.getByPlaceholderText(/SEARCH/i);

    //fireEvent.change(searchBar, { target: { value: 'MARVEL' } });
    userEvent.type(searchBar, 'MARVEL');

    await waitFor(
      () => {
        expect(screen.queryByText('Marvel Apes')).toBeInTheDocument();
        expect(screen.queryByText('Marvel Boy')).toBeInTheDocument();
        expect(screen.queryByText('Marvel Zombies')).toBeInTheDocument();
      },
      { timeout: 3000 }
    );

    userEvent.type(searchBar, '{backspace}{backspace}{backspace}{backspace}');

    await waitFor(
      () => {
        expect(screen.queryByText('Marvel Apes')).toBeNull();
        expect(screen.queryByText('Marvel Boy')).toBeNull();
        expect(screen.queryByText('Marvel Zombies')).toBeNull();
        expect(screen.queryByText('3-D Man')).toBeInTheDocument();
      },
      { timeout: 3000 }
    );
  });
});
