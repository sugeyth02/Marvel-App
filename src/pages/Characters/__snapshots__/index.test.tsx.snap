// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Characters list view should renders a list of characters initially 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <a
          class="header__link"
          href="/"
        >
          <img
            alt="Marvel logo"
            class="header_img"
            src="/img/Marvel_Logo.svg.png"
          />
        </a>
        <div
          class="buttons"
        >
          <button
            class="button__hidden"
          >
            <i
              class="fa-solid fa-eye"
            />
          </button>
          <a
            href="/bookmark"
          >
            <i
              class="fa-solid fa-bookmark"
            />
          </a>
        </div>
      </div>
      <ul
        class="pages"
      >
        <li
          class="pages__link"
        >
          <a
            href="/characters"
          >
            CHARACTERS
          </a>
        </li>
        <li
          class="pages__link"
        >
          <a
            href="/comics"
          >
            COMICS
          </a>
        </li>
        <li
          class="pages__link"
        >
          <a
            href="/stories"
          >
            STORIES
          </a>
        </li>
      </ul>
    </div>
    <div
      class="outlet"
    >
      <div
        class="container"
      >
        <h1
          class="container__header"
        >
          MARVEL CHARACTERS LIST
        </h1>
        <div
          class="searchBar"
        >
          <button
            class="searchBar_button"
          >
            <i
              class="fa-solid fa-magnifying-glass"
            />
          </button>
          <label
            class="searchBar__label"
            for="search"
          >
            search
          </label>
          <input
            class="searchBar__input"
            name="search"
            placeholder="SEARCH"
            type="text"
          />
        </div>
        <div
          class="container"
        >
          <div
            class="container__filters"
          >
            <div
              class="filters__select"
            >
              <label
                for="comics"
              >
                Comics:
              </label>
              <select
                id="comics"
                name="comics"
              />
            </div>
            <div
              class="filters__select"
            >
              <label
                for="stories"
              >
                Stories:
              </label>
              <select
                id="stories"
                name="stories"
              />
            </div>
          </div>
          <div
            class="container__cards"
          >
            <div
              class="container"
            >
              <div
                class="container__image"
                style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784/portrait_fantastic.jpg);"
              />
              <div
                class="container__info"
              >
                <h1>
                  3-D Man
                </h1>
                <div
                  class="buttons"
                >
                  <button
                    class="container__hide"
                  >
                    <i
                      class="fa-solid fa-eye-slash"
                    />
                  </button>
                  <button
                    class="container__bookMark "
                    data-testid="bookmark"
                  >
                    <i
                      class="fa-solid fa-bookmark"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="container__image"
                style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg);"
              />
              <div
                class="container__info"
              >
                <h1>
                  Alain
                </h1>
                <div
                  class="buttons"
                >
                  <button
                    class="container__hide"
                  >
                    <i
                      class="fa-solid fa-eye-slash"
                    />
                  </button>
                  <button
                    class="container__bookMark "
                    data-testid="bookmark"
                  >
                    <i
                      class="fa-solid fa-bookmark"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <button
              class="container__button"
              disabled=""
            >
              <i
                class="fa-solid fa-angle-left"
              />
            </button>
            <div
              class="numbers"
            />
            <button
              class="container__button"
            >
              <i
                class="fa-solid fa-angle-right"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <img
        alt="Marvel logo"
        class="container__img"
        src="/img/m_logo.png"
      />
      <p
        class="container__reserved"
      >
        Data provided by Marvel. Â© 2014 Marvel
      </p>
      <div
        class="container__info"
      >
        <a
          class="info__link"
          href="https://help.marvel.com/hc/en-us"
        >
          Contact Us
        </a>
      </div>
    </div>
  </div>
</div>
`;
